### PR TITLE
SDL325: Compile with defenses enabled

### PR DESCRIPTION
**Pull request by sfvaroglu**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +173,25 @@ if(USE_NGRAPH AND NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-comment")
   set(NGRAPH_INSTALL_PREFIX ${CMAKE_BINARY_DIR}/3rdparty/ngraph-mxnet-bridge)
 
+  # Flags required for SDL325
```
Comment (cconvey) [inline CMakeLists.txt]: Is "SDL 325" meaningful outside of Intel?

File: CMakeLists.txt [hunk 2]
```diff
@@ -173,6 +173,25 @@ if(USE_NGRAPH AND NOT MSVC)
...
+      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
+      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
+  endif()
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+      if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.9)
+          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector")
+      else()
+          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector-strong")
+      endif()
+  endif()
```
Comment (cconvey) [inline CMakeLists.txt]: It looks like Clang is [a supported compiler for MXnet](https://mxnet.incubator.apache.org/install/build_from_source.html#c-build-tools).  Are we okay with these compiler flags _not_ being applied to Clang builds on Linux?

File: CMakeLists.txt (hunk 1 above)
Comment (sfvaroglu) [reply CMakeLists.txt]: No, but it's meaningful to us. Do you prefer an explanation (e.g. compile with defenses enabled) instead?
